packages/thirdweb/src/wallets/in-app/native/helpers/wallet/retrieval.ts

```diff
@@ -58,9 +58,27 @@ async function getAccountFromShares(args: {
   shares: string[];
 }): Promise<Account> {
   const { client, shares } = args;
+  const privateKey = await (async () => {
+    try {
+      return await getWalletPrivateKeyFromShares(shares);
+    } catch (e) {
+      // If the private key reconstruction fails, try to reset the device share and prompt the user to try again
+      // This can happen if a user's account has been migrated or otherwise modified in the backend to use a new wallet. In that case, we need to reset their device state to get a new share
+      if (
+        e instanceof Error &&
+        e.message === "Invalid private key reconstructed from shares"
```

Comment:
would be good to put this string in a const so we dont change it accidentally and forget to update here